### PR TITLE
Feature/clubhouse

### PR DESCRIPTION
**Pull request by Gracetemitope**
In this branch, I made use of Devise gem for Authentication (sign up and Login). Also, as part of the project requirement,  this app only allows authenticated members to create and see post details when guest users only view posts.

### CODE REVIEW

**Comment by mke2111** [comment]
## STATUS: INVALID 🔴 

### Hey @Gracetemitope, 👋🏽

I'm invalidating this PR simply because the GA is failing. 
This is probably brought about by the ruby version you are running clashing with the linters setup by microverse. Kindly visit the [questions](https://questions.microverse.org/) for a solution. A solution exists

**Comment by mke2111** [comment]
## STATUS: INVALID 🔴 

### Hey @Gracetemitope, 👋🏽

I'm invalidating this PR simply because the GA is failing. 
This is probably brought about by the ruby version you are running clashing with the linters setup by microverse. Kindly visit the [questions](https://questions.microverse.org/) for a solution. A solution exists

**Review by MahmoudBakr23 — COMMENTED** [review]
# **Invalid Submission 🛑**

<hr>

Hello there :)
Please keep in mind that in order to get your PR reviewed, it has to be professionally set up. This includes the correct Linter, PR summary, and a descriptive README file. Kindly add a description for your PR and submit for another review.

<hr>

Good luck and happy coding. 👨‍💻💻

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @Gracetemitope   ! :wave: :wave:

#### You have done well so far :clap: However, there are changes required for you to make in order for this project to be approved. Kindly work on them and submit them for another code review.


**Recommended**
- [x] I could not run your app locally, I got this error whenever I tried to mgrate the database:
```
Caused by:
NoMethodError: undefined method `tex' for #<ActiveRecord::ConnectionAdapters::SQLite3::TableDefinition:0x000055ed5d723f08>
Did you mean?  text
```
- [x] It creates the application routes using resources :posts, only: [:new, :create, :index]  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

- [x] it initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Review by smcommits — CHANGES_REQUESTED** [review]
# Status: Changes Required 

## Hello, @Gracetemitope! You have done a great job. The application appears to be amazing. To make this project even better, we still need to address some issues that the previous code reviewer pointed out. 

## Required Changes

#### You still haven't addressed the following issue that was pointed out by the previous code reviewer.

- [ ] **The Post resource should only have an index, new, and create RESTful routes.**

You only need to create an endpoint for index, new, and create for the Post resource. You achieve this using a specific method in your routes.rb file. Please implement it as it is a mandatory project requirement. 


#
## Please reach out to me on Slack if you need any assistance.

**Comment by mimipeshy** [comment]
## STATUS: CHANGES REQUESTED 🔴 ♻️ 
Hello @Gracetemitope 👋🏾 👋🏾 , great job so far on this milestone. I like that you have tried to implement some feedback given by the TSE's. Your linter checks pass as well
However, we still need to work on the following before approval ⬇️ 

- [ ] Kindly check out my comments on what  needs to be worked on 😄 

That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

File: app/models/user.rb
Comment (mimipeshy) [inline app/models/user.rb]: - [x] Remove all commented code from your project

File: app/views/layouts/application.html.erb
Comment (mimipeshy) [inline app/views/layouts/application.html.erb]: - [x] Remove any commented code from your project

File: config/routes.rb
Comment (mimipeshy) [inline config/routes.rb]: - [x] Kindly remove all extra spaces in this file
- [x] As the two previous TSE's noted ,the Post resource should only have an index, new, and create RESTful routes. i.e
`resources :posts, only: [:new, :create, :index]`  as described here
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - ✅ The code looks clean and professional. 💯
  - ✅ The project follows Git flow. 💯

- ### To Improve
  - [ ] [MANDATORY] It seems like you did not implement any of the previous required changes. kindly work on them and then request another re-review.   

_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Comment by Gracetemitope** [comment]
Apologies. I didn't know I didn't push from my local machine.

**Review by rukundoeric — APPROVED** [review]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - ✔️ Follows gitwork-flow 💯
  - ✔️ Codebase is clean and professianl 💯
  - ✔️ The app is functional 💯

- ### To Improve
  - n/a

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**
